src/common/snippets/src/generator.cpp

```diff
@@ -14,25 +14,28 @@
 namespace ov {
 namespace snippets {
 
-LoweringResult Generator::generate(lowered::LinearIR& linear_ir, const void* compile_params) const {
+LoweringResult Generator::generate(const lowered::LinearIRPtr& linear_ir, const void* compile_params) const {
     OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::Generator::generate")
 
+    // Before code gen we have to reset KernelExecutor Table - it should be empty
+    target->get_runtime_configurator()->reset_kernel_executor_table();
```

Comment:
> Fixed not-first code generation for the same Subgraph node - kernel executor table should be reset

Do I understand correctly that this is the mentioned fix? Should it be merged in scope of #25638?

Reply:
Yeah, it fixes non-first code generation.
Just imagine we have one Subgraph with dynamic MHA inside.
First infer request: `input_static_shape0` -> `broadcsting_mask0` -> code generation -> fill `KernelExecutorTable`
Second infer request: `input_static_shape1` -> `broadcsting_mask1` (broadcasting pattern is new) -> code generation with filed `KernelExecutorTable` from the previous infer request, however for each code gen it should new and unique.
I'd say that each binary code has own `KernelExecutorTable` -> so before code gen we should reset the table.

It was observed in this feature development so I decided to fix it here

Reply:
Thanks for the explanation!